clang/lib/Frontend/CompilerInvocation.cpp

```diff
@@ -3426,6 +3476,23 @@ static bool ParseFrontendArgs(FrontendOptions &Opts, ArgList &Args,
     }
   }
 
+  if (!Opts.CASInputFileCacheKey.empty()) {
+    std::optional<llvm::MemoryBufferRef> Buff;
+    if (llvm::Error E = determineInputFromCacheKey(Opts.CASInputFileCacheKey,
+                                                   CASOpts, Diags, Buff)
+                            .moveInto(Opts.CASInputFileID)) {
+      Diags.Report(diag::err_fe_unable_to_load_input_cache_key)
+          << Opts.CASInputFileCacheKey << llvm::toString(std::move(E));
+    } else {
+      Inputs.push_back(Buff->getBufferIdentifier().str());
+
+      FrontendInputFile &InputFile = Opts.Inputs.back();
+
+      InputFile =
+          FrontendInputFile(*Buff, InputFile.getKind(), InputFile.isSystem());
```

Comment:
So the idea here is that you have a filename passed as the input and that determines the "kind" of input we're handling?

Reply:
No, the input handling code works very similarly to how `-fcas-include-tree` works: there are no input files on the command line, so this function manufactures one (stdin) and applies `-x` on it. Here we just take that manufactured `FrontendInputFile` and fill in the buffer (specified by `-fcas-input-file-id`).